### PR TITLE
travis: use ccache on Linux and macOS

### PR DESCRIPTION
**Pull request by julianoes**
I think it is worth a try to speed up the build.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -80,6 +81,10 @@ install:
     docker run --rm dockcross/linux-armv7 > ./dockcross-linux-armv7;
     chmod +x ./dockcross-linux-armv7;
   fi
+- if [[ "${BUILD_TARGET}" = "osx_build" ]] || [[ "${BUILD_TARGET}" = "ios_build" ]] || [[ "${BUILD_TARGET}" = "ios_simulator_build" ]]; then
+    brew install ccache;
+    export PATH="/usr/local/opt/ccache/libexec:$PATH";
```
Comment (JonasVautherin) [inline .travis.yml]: so that's a travis path that stays between builds?

Comment (julianoes) [reply .travis.yml]: According to the docs, yes. I doubt it though.
https://docs.travis-ci.com/user/caching/#ccache-cache

File: CMakeLists.txt [hunk 2]
```diff
@@ -12,6 +12,23 @@ endforeach()
 
 project(mavsdk_superbuild)
 
+# ccache
```
Comment (JonasVautherin) [inline CMakeLists.txt]: Why are we moving that here? Does it affect the superbuild?

Comment (julianoes) [reply CMakeLists.txt]: I was hoping that it would enable ccache for the superbuild dependencies.

Comment (JonasVautherin) [reply CMakeLists.txt]: I don't understand completely how this cache config works in cmake, but it seems to be just about setting a few variables. Those would have to be propagated to the dependencies (I would say that they are not in the current setup)

Comment (julianoes) [reply CMakeLists.txt]: You're right 🤔 .

**Comment by julianoes** [comment]
We moved from travis-ci to GitHub workflows now, closing.

**Pull request closed without merging**